routing.go

```diff
@@ -680,30 +687,16 @@ func (dht *IpfsDHT) FindPeer(ctx context.Context, id peer.ID) (_ peer.AddrInfo,
 		return peer.AddrInfo{}, err
 	}
 
+	// refresh the cpl for this key as the query was successful
+	if ctx.Err() == nil {
+		kadID := kb.ConvertPeerID(id)
+		dht.routingTable.ResetCplRefreshedAtForID(kadID, time.Now())
+	}
+
```

Comment:
Why no shortcut check? This query aborts if the peer happens to connect to us totally independently of the DHT.

Reply:
Done.